app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMap.java

```diff
@@ -92,6 +96,16 @@ protected void drawDevice(Canvas canvas) {
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, strokeRadius, whiteFillPaint);
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, strokeRadius, secondaryStrokePaint);
         canvas.drawCircle(deviceCenter.x, deviceCenter.y, pixelsPerDip * 8, secondaryFillPaint);
+
+
+        if (deviceLocationPredictor.hasPrediction) {
+            drawDeviceLocationPrediction(canvas, getPointFromLocation(deviceLocationAnimator.getLocation()));
+        }
+    }
+
+    protected void drawDeviceLocationPrediction(Canvas canvas, PointF deviceCenter) {
```

Comment:
Why do you need to pass the `deviceCenter`? You can directly use `getPointFromLocation(deviceLocationAnimator.getLocation())` in the method.